include/networkit/numerics/LinearSolver.hpp

```diff
@@ -84,6 +85,31 @@ class LinearSolver {
      * @note If the solver does not support parallelism during solves, this function falls back to solving the systems sequentially.
      */
     virtual void parallelSolve(const std::vector<Vector>& rhs, std::vector<Vector>& results, count maxConvergenceTime = 5 * 60 * 1000, count maxIterations = std::numeric_limits<count>::max());
+
+
+    /**
+     * RHS loader. Accepts index of RHS column and mutable reference to properly sized row Vector, which is not necessary zero-valued.
+     * Returns a reference to RHS vector.
+     */
+    using RHSLoader = std::function<Vector&(count, Vector&)>;
+
+    /**
+     * Result processor. Accepts index of result column and immutable reference to result Vector.
+     */
+    using ResultProcessor = std::function<void(count, const Vector&)>;
```

Comment:
Instead of using `std::function` here (which has a high overhead), I would turn the parallelSolve function into a template (that requires that we implement it in the header which is a bit annoying, but not an issue).

Reply:
there is a problem with ```parallelSolve``` being virtual. I thought of doing type-erasure but that would require std::any, which only appeared in C++17

Reply:
Hm, one option would be to not make it virtual. It is a new overload anyway, so maybe that is a reasonable trade-off?

Reply:
Callers that do not know the type of the solver that they are using could still use the old `parallelSolve`.

Reply:
> Hm, one option would be to not make it virtual. It is a new overload anyway, so maybe that is a reasonable trade-off?

I think the same. Also there seems to be only one place where LinearSolver polymorphism is used (MaxentStress), so in most cases callers will know the type of solver.